### ISSUE TITLE
Nathaniel Howe: The generated savegame will ignore the given inputs. 

### ISSUE THREAD

**Issue by manavortex** [gibbed/Gibbed.DragonAge.SaveGenerator#1]
Hey, 
my friend got in touch on twitter, but github wouldn't let her register. She has been the one to do the testing (and she was incredibly thorough), I handled the code side of things. :)

Input: 
Recruited Nathaniel         checked
NATHANIEL_FRIENDLY_ELIGIBLE checked
Nathaniel died in the siege unchecked

Expected behaviour: The quest [Finding Nathaniel](https://dragonage.fandom.com/wiki/Finding_Nathaniel) is available in Act3 of DA2
Observed behaviour: The quest [Fool's Gold](https://dragonage.fandom.com/wiki/Fool's_Gold) is available in Act3 of DA2 

I tried to change [this](https://github.com/gibbed/Gibbed.DragonAge.SaveGenerator/blob/aab299ed9ceae0424283b64d12f26105d9ba1d63/projects/Gibbed.DragonAge.SaveGenerator/MainWindow.xaml.cs#L936) from 22 to 12 (after the workaround [here](https://dragonage.fandom.com/wiki/Finding_Nathaniel#Bugs)), but the generated savegame still triggered Fool's Gold. 

toggling "died in the siege" instead of "NATHANIEL_FRIENDLY_ELIGIBLE" also didn't fix it, I had hoped for a gui label mixup

**Comment by gibbed** [comment]
The flags here do appear to be what appear in the worldvault, which is how Dragon Age 2 handles imports:

```
42,dao,0,cod_cha_nathaniel,struct,,$PLOTS[$PLOT_GUIDS:"F29B4323C13049BABE1D2E3007C6AD1D"]
57,dao,0,gxa000pt_party,struct,,$PLOTS[$PLOT_GUIDS:"E3F7CA9BE9C543378764CD4D5AFC00A3"]
58,dao,0,gxa000pt_app_nathaniel,struct,,$PLOTS[$PLOT_GUIDS:"5DABBE1A8F3A440592A9F3078D3ABE16"]
140,dao,2065,GXA_NATHANIEL_RECRUITED,bool,0,$gxa000pt_party#9
159,dao,2084,COD_CHA_NATHANIEL_DIED_IN_VGK_SIEGE,bool,0,$cod_cha_nathaniel#12
160,dao,2085,GXA_APP_NATHANIEL_FRIENDLY_ELIGIBLE,bool,0,$gxa000pt_app_nathaniel#22
```

Would you happen to know if a real save from Dragon Age with Nathaniel still alive imports correctly for this?

**Comment by gibbed** [comment]
Additonally, I'm going to redownload DA2 and see if they ended up patching the worldvault at some point. Maybe this is a bug they fixed and my save generator simply was never updated to compensate.

**Comment by gibbed** [comment]
It doesn't look like they ever patched the worldvault file, so I'm guessing there's more requirements for that quest to appear or it's bugged.

[This mod on NexusMods ("Import Vault Fixes")](https://www.nexusmods.com/dragonage2/mods/4236) appears to suggest that the Finding Nathaniel quest line is bugged in general.

**Comment by manavortex** [comment]
Hey gibbed,
thank you for looking at this - there's a console command that you run in Origins to fix the generated savegame for DA2, and according to the person who tested all this, that's working: 
https://dragonage.fandom.com/wiki/Finding_Nathaniel#Bugs

Does that help anything?

**Comment by gibbed** [comment]
What that command should be doing is unsetting this flag:

https://github.com/gibbed/Gibbed.DragonAge.SaveGenerator/blob/aab299ed9ceae0424283b64d12f26105d9ba1d63/projects/Gibbed.DragonAge.SaveGenerator/MainWindow.xaml.cs#L959-L962

**Comment by manavortex** [comment]
My attempt at "fixing" this was to try and directly add in whatever the command did to the savegame editor, but that didn't work.

**Comment by gibbed** [comment]
Yeah, what I mean is that command in the link you've referenced, would unset the plot flag that is set by that code I linked in my previous comment.

That plot flag shouldn't be getting set by the save generator when it's unchecked, at least.

**Comment by gibbed** [comment]
What I would ask going forward is, can you provide me a copy of the save that's generated that's broken, then a copy of the save that's been modified by that command?

You can attach them to this issue (probably need to ZIP them first though).

**Comment by VicFromTheFade** [comment]
> Would you happen to know if a real save from Dragon Age with Nathaniel still alive imports correctly for this?

Yes a real save imports correctly.

**Comment by gibbed** [comment]
Wondering now if this is possibly a bug due to default values being provided by the game on import when plot flags are entirely unset.

**Comment by VicFromTheFade** [comment]
One is the normal save, the other is generated. Nathaniel was marked as alive in both occasions. Hopefully it will help shed some light.

[Duran.zip](https://github.com/gibbed/Gibbed.DragonAge.SaveGenerator/files/4809722/Duran.zip)
[Generated_Duran.zip](https://github.com/gibbed/Gibbed.DragonAge.SaveGenerator/files/4809723/Generated_Duran.zip)

**Comment by gibbed** [comment]
Is that generated save one that was modified by the console command, or unmodified from the save generator?

**Comment by VicFromTheFade** [comment]
Unmodified. I generated it and immediately uploaded it here.

**Comment by VicFromTheFade** [comment]
Hey, do you have any other questions, I might be able to help you with?

**Comment by Patchier** [comment]
Hi gibbed, I know it's been...a little while since this was commented on, but curious if you ever figured out the issue? I got the urge to play DA2 again, but not super keen on playing through Origins all over again.
